### PR TITLE
[Isolation Protocol] Add Bionic Vampire class

### PR DESCRIPTION
**Pull request by Standing-Storm**
#### Summary
Mods "[Isolation Protocol] Add Bionic Vampire class"

#### Purpose of change

Had the idea for two new classes for the mod. This is the first of them.

#### Describe the solution

Add the Bionic Vampire class.  This is not a vampire with bionics, or Magiclysm's weirdo who uses bionics to try to be a vampire. They're an assassin packed to the gills with metal, whose faulty bionics drive them to kill to gain more bionic power.

- Every second they lose some bionic power, unless they're in a safe area (one of the rest point safe areas, not just an area with no visible enemies)
- Killing enemies in melee (NOT at range) restores bionic power, scaling with the difficulty of the enemy.
- Reaching a safe level also restores bionic power.
- They're so full of cybernetics they have no room to install any other CBMs, even if you find them (done by simply preventing new CBM installs, so that if you find a power-generating CBM you can't just use it to avoid the whole point of the class)

Gameplay is thus a balance of killing enemies in melee to restore power, using that power to fight more dangerous enemies, and shooting enemies that are just too dangerous.

#### Testing

Played a few games. Keeping your bionic power up is a struggle, especially against dangerous opponents, but not impossible.

#### Additional context

My second class idea is a wizard, but with what I think is a pretty unique twist on the concept. Stay tuned.

### CODE REVIEW

File: data/mods/Isolation-Protocol/Player/Perks/perks.json [hunk 1]
```diff
@@ -60,5 +60,16 @@
     "enchantments": [ "iso_ench_geas_buff" ],
     "purifiable": false,
     "valid": false
+  },
+  {
+    "type": "mutation",
+    "id": "iso_bionic_vampire",
+    "name": { "str": "Power Drain" },
+    "points": 0,
+    "description": "Your cybernetic implants are constantly losing power, but a quirk of their design allows you regain power through murder.\n\nYou lose bionic power every second, but melee kills restore power, scaling by with the danger level of your target.",
```
Comment (TheShadowFerret) [inline data/mods/Isolation-Protocol/Player/Perks/perks.json:69]:
```suggestion
    "description": "Your cybernetic implants are constantly losing power, but a quirk of their design allows you regain power through murder.\n\nYou lose bionic power every second, but melee kills restore power, scaling with the danger level of your target.",
```

**Comment by John-Candlebury** [comment]
I won't be available for actual review but I approve the concept.

File: data/mods/Isolation-Protocol/Player/Perks/perk_eoc.json [hunk 2]
```diff
@@ -84,5 +84,55 @@
...
+    "id": "EOC_ISO_BIO_VAMPIRE_RESTORE",
+    "eoc_type": "EVENT",
+    "required_event": "character_kills_monster",
+    "condition": { "and": [ { "math": [ "u_bionic_in_melee", "==", "1" ] }, { "u_has_trait": "iso_bionic_vampire" } ] },
+    "//": "* in math is so the number before the * is kJ of bionic power restored",
+    "effect": [
+      { "math": [ "u_val('power')", "+=", "1.75 * 1000000" ] },
+      { "if": { "npc_has_flag": "PUSH_VEH" }, "then": { "math": [ "u_val('power')", "+=", "50 * 1000000" ] } },
+      {
+        "if": { "npc_has_flag": "CORNERED_FIGHTER " },
```
Comment (GuardianDll) [inline data/mods/Isolation-Protocol/Player/Perks/perk_eoc.json:99]: trailing spaces strike back

File: data/mods/Isolation-Protocol/Player/Perks/perk_eoc.json [hunk 3]
```diff
@@ -84,5 +84,55 @@
...
+        "then": { "math": [ "u_val('power')", "+=", "0.5 * 1000000" ] }
+      },
+      {
+        "if": { "npc_has_flag": "PRIORITIZE_TARGETS  " },
```
Comment (GuardianDll) [inline data/mods/Isolation-Protocol/Player/Perks/perk_eoc.json:103]: and here

**Pull request merged**